ecadlabs/taqueria
# Install Taqueria from VS Code Extension

# 🏗️ PR ➾ Install Taqueria from VS Code Extension

Fixes #106 (issue)

## 🪂 Pre-Merge Checklist

- [ ] 🦀 Automated tests have been written and added to this PR
- [ ] 🤿 Corresponding changes have been made to all documentation
- [x] 🏖️ New and existing unit tests pass locally and in CI
- [ ] 🔱 The test plan has been implemented and verified by an SDET
- [ ] 🏄‍♂️ Myself and others have built this branch and done manual testing on the change
- [ ] ⛵ A summary describing the change has been written for release notes

----------------------------------------------------------------------------------------------------------------------------

## 🪁 Description

This PR adds ability (a button) to install taq CLI automatically. It downloads the binary, makes it executable and copy to user bin path to make it reachable in terminal.

## 🎢 Test Plan

1. Remove taq from your system
2. Lanuch vscode with extension
3. It should display no taq installed info and an install button
4. Click on the button
5. It should display progress info, then ask for sudo password
6. It should display notification all was ok installed and the left side panel should show taq is available
7. Repeat and on step 5 provide wrong password
8. It should display warning message and output errors from commands in taqueria logs window

----------------------------------------------------------------------------------------------------------------------------

### 🛸 Type of Change

- [ ] 🐟 Minor change (non-breaking change of very limited scope)
- [ ] 🐳 Major refactor (breaking changes and significant impact to the codebase)

----------------------------------------------------------------------------------------------------------------------------
